src/logic/calculate.js

```diff
@@ -1,27 +1,65 @@
 /* eslint-disable import/extensions */
 /* eslint-disable prefer-const */
+/* eslint-disable use-isnan */
+/* eslint-disable no-restricted-globals */
 
 import operate from './operate.js';
 
 const calculate = (calc, name) => {
-  let { total, next, operation } = calc;
-  switch (name) {
-    case '+/-':
-      total *= -1;
-      next *= -1;
-      break;
-    case '.':
-      total = parseFloat(`${`${total}.${next}`}`);
-      break;
-    case 'AC':
-      total = 0;
-      next = 0;
-      break;
-    default:
-      total = operate(total, next, operation);
-      break;
+  const {
+    total, next, operation, result,
+  } = calc;
+  let newTotal = total;
+  let newNext = next;
+  let newOp = operation;
+  let newResult = result;
+  if (!isNaN(name)) {
+    if (newTotal.length === 0 || newTotal.toString().slice(-1) === '.') {
+      newTotal = `${newTotal}${name}`;
+      newResult = `${newResult}${name}`;
+    } else {
+      newNext = `${newNext}${name}`;
+      newResult = `${newResult}${name}`;
+    }
+  } else {
+    switch (name) {
+      case '+/-':
+        newTotal *= -1;
+        newNext *= -1;
+        newResult = newTotal;
+        break;
```

Comment:
:bug: The `+/-` button is not working as it should.

The sign is not capturing correctly the values that are intended to alternate at times.

For example, when two numbers (or more) are clicked neither `total` nor `next` keeps track of these values. Instead, `result` seems to be the one keeping track of the changes. Here `result` is overwritten by `total` that is not keeping track of the value.
So when `55` is displayed for example it turns to `-5` when the `+/-` button is clicked.

See image below to see step to reproduce the bug. (You can also log the `state` value in the `render` method of the `App` component for more insight)

![cal-error2](https://user-images.githubusercontent.com/36057474/71619351-53ef5680-2bc4-11ea-96ed-d528b504dceb.gif)

Reply:
Great catch! It's now fixed